File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const hash = [];
    participant.forEach(p => {
        hash[p] = hash[p] ? hash[p] + 1 : 1;
    })
    completion.forEach(p => {
        hash[p]--;
    })
    for(let key in hash) {
        if( hash[key] >= 1) return key;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    d = {}
    
    for p in completion:
        if p in d:
            new_c = d[p] + 1
            d[p] = new_c
        else:
            d[p] = 1
        
    for p in participant:
        if p in d:
            new_c = d[p] - 1
            if new_c <= 0:
                del d[p]
        
            else:
                d[p] = new_c
        else:
            return p
        